### DISCUSSION TITLE
How to upgrade all except "xxxx"

### DISCUSSION THREAD

**Discussion by Ango1095** [microsoft/winget-cli#5021] [Q&A]
Hello
how to put an exception in the upgrade all command
I would like to update everything except 1 software
Thank's

**Comment by mdanish-kh** [comment]
You can pin that package using `winget pin add <PackageId>` and then run `winget upgrade --all`

**Reply by mdanish-kh** [reply]
There isn't a way to do this inline with the `winget upgrade` command though. You'll have to remove that pin later if you want to install upgrades for the pinned package

**Comment by Ango1095** [comment]
Thank you for your quick and efficient help.
